HARK/ConsumptionSaving/ConsMarkovModel.py

```diff
@@ -321,29 +321,26 @@ def calc_EndOfPrdvPP(self):
             End-of-period marginal marginal value of assets at each value in
             the grid of assets.
         """
-        def vpp_next(shocks, a_nrm):
-            return shocks[0] ** (- self.CRRA - 1.0) \
-                * self.vPPfuncNext(self.m_nrm_next(shocks, a_nrm))
+
+        def vpp_next(shocks, a_nrm, Rfree):
+            return shocks[0] ** (-self.CRRA - 1.0) * self.vPPfuncNext(
+                self.m_nrm_next(shocks, a_nrm, Rfree)
+            )
 
         EndOfPrdvPP = (
             self.DiscFacEff
             * self.Rfree
             * self.Rfree
             * self.PermGroFac ** (-self.CRRA - 1.0)
-            * calc_expectation(
-                self.IncShkDstn,
-                vpp_next,
-                self.aNrmNow
-            )
+            * calc_expectation(self.IncShkDstn, vpp_next, self.aNrmNow, self.Rfree)
```

Comment:
This looks like it's using the old calc_expectation construction. A reason for that?

Reply:
I only set up the new method for ConsIndShockModel, this was just some downstream clean up I had to do. We can discuss whether we should implement this across all of HARK at the next meeting.